src/main/java/io/codelex/flightplanner/maintree/services/FlightServiceDatabase.java

```diff
@@ -0,0 +1,132 @@
+        Flight flight = new Flight(input);
+        flight.setFrom(airportFrom);
+        flight.setTo(airportTo);
+
+        ExampleMatcher flightMatcher = ExampleMatcher.matching()
+                .withIgnorePaths("id")
+                .withMatcher("airport_from", ignoreCase())
+                .withMatcher("airport_to", ignoreCase())
+                .withMatcher("carrier", ignoreCase())
+                .withMatcher("departure_time", ignoreCase())
+                .withMatcher("arrival_time", ignoreCase());
+        Example<Flight> example = Example.of(flight, flightMatcher);
+
+        if (flightRepository.exists(example)) {
+            throw new ResponseStatusException(HttpStatus.CONFLICT);
+        }
+
+        if ((input.getFrom().equals(input.getTo())) || (flight.getArrivalTime().isBefore(flight.getDepartureTime())) ||
+                (flight.getArrivalTime().equals(flight.getDepartureTime()))) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST);
+        }
+
+        return this.flightRepository.save(flight);
+
+    }
+
+
+    public synchronized Flight getFlight(int input) {
+        return this.flightRepository.findById(input)
+                .orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND, "Flight not found"));
+    }
+
+
+    public synchronized void removeFlight(int id) {
+        Flight flightToDelete = this.flightRepository.findById(id)
+                .orElseThrow(() -> new ResponseStatusException(HttpStatus.OK));
+        this.flightRepository.delete(flightToDelete);
+    }
+
+
+    public HashSet<Airport> searchAirports(String input) {
+        HashSet<Airport> foundAirports = new HashSet<>();
+        Iterable<Flight> flightList = this.flightRepository.findAll();
+
+        String inputConverted = input.toLowerCase().trim();
+        for (Flight flight : flightList) {
+            String fromCountry = flight.getFrom().getCountry().toLowerCase();
+            String fromCity = flight.getFrom().getCity().toLowerCase();
+            String fromAirport = flight.getFrom().getAirport().toLowerCase();
+            String toCountry = flight.getTo().getCountry().toLowerCase();
+            String toCity = flight.getTo().getCity().toLowerCase();
+            String toAirport = flight.getTo().getAirport().toLowerCase();
+
+            if ((fromCountry.startsWith(inputConverted)) || (fromCity.startsWith(inputConverted)) ||
+                    (fromAirport.startsWith(inputConverted)) || (toCountry.startsWith(inputConverted)) ||
+                    (toCity.startsWith(inputConverted)) || (toAirport.startsWith(inputConverted))) {
+                foundAirports.add(flight.getFrom());
+            }
+
+        }
+        return foundAirports;
+    }
+
+
+    public PageResults<Flight> searchFlights(FlightSearch input) {
+        List<Flight> resultsList = new ArrayList<>() {
+        };
+
+        Iterable<Flight> flightList = flightRepository.findAll();
```

Comment:
No findAll ir jāuzmanās, ja DB ir 100 000 ieraksti, tad šis ielasīs tos visus atmiņā un visticamāk sistēmu nokārs - tā vietā būtu jāmeklē DB pusē. To var darīt tāpat kā darīji augstāk ar to example.